subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/ReportFriendlyFailureHandler.kt

```diff
@@ -96,7 +97,13 @@ class ReportFriendlyFailureHandler(targetContext: Context) : FailureHandler {
                 e is AssertionFailedError -> {
                     throw AssertionFailedError(e.normalizedMessage())
                 }
-                e is NoMatchingViewException -> {
+                e is NoMatchingRootException -> {
+                    throw UITestFrameworkException(
```

Comment:
This is a mistake. We shouldn't wrap espresso error by UITestFrameworkException.
By default, retry skips Espresso errors but not UITestFrameworkException.

One possible solution is to check the cause of UITestFrameworkException.